### PR TITLE
mgr/dashboard_v2: Add cdPasswordButton directive

### PR DESCRIPTION
**Pull request by votdev**
bildschirmfoto vom 2018-02-01 20-13-16

Signed-off-by: Volker Theile <vtheile@suse.com>

### CODE REVIEW

**Review by ricardoasmarques — COMMENTED** [review]
Does it make sense to create a dedicated  module for directives, inside shared module?

I’m just asking because we have open PRs that are creating the components module inside shared.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/cd-password-button.directive.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { Directive, ElementRef, Input, HostListener, Renderer2 } from '@angular/core';
+
+@Directive({
+  selector: '[cd-password-button]'
+})
+export class CdPasswordButtonDirective {
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/shared/cd-password-button.directive.ts]: Class name should not contain the application prefix.

**Review by tspmelo — CHANGES_REQUESTED** [review]
I agree with @ricardoasmarques , otherwise we would end up with lots of files inside shared folder.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/cd-password-button.directive.ts (hunk 1 above)
Comment (votdev) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/shared/cd-password-button.directive.ts]: Fixed

**Comment by votdev** [comment]
The directive has been relocated into the new app/shared/directives  directory.

**Review by ricardoasmarques — DISMISSED** [review]
lgtm

**Review by LenzGr — DISMISSED** [review]

**Review by tspmelo — DISMISSED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/directives/password-button.directive.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  }
+
+  @HostListener('click')
+  onClick() {
+    // Modify the type of the input field.
+    this.inputElement.type = (this.inputElement.type === "password") ? "text" : "password";
+    // Update the button icon/tooltip.
+    this.update();
+  }
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/shared/directives/password-button.directive.ts]: Isn't it possible to use a template for this? With a template we don't have to add and remove classes and it's easier to read.

`<i #pwicon class="icon-prepend fa" [ngClass]="{'fa-eye':!isText(), 'fa-eye-slash':isText()}" (click)="changeType()">`

**Review by Devp00l — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/shared/directives/password-button.directive.ts (hunk 2 above)
Comment (votdev) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/shared/directives/password-button.directive.ts]: Didn't find a way to do so.

**Comment by jecluis** [comment]
@votdev is there anything holding up this PR?

**Review by ricardoasmarques — DISMISSED** [review]

**Review by tspmelo — CHANGES_REQUESTED** [review]
Please fix the lint and unit test erros

**Comment by votdev** [comment]
Lint and unit test issues are fixed. Can be merged now.

**Review by tspmelo — APPROVED** [review]
lgtm

**Review by Devp00l — APPROVED** [review]

**Pull request merged**